WcaOnRails/app/webpacker/lib/hooks/useRequest.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { fetchWithAuthenticityToken } from '../requests/fetchWithAuthenticityToken';
+
+export default function useRequest(successTrigger = null, errorTrigger = null) {
+  const [loading, setLoading] = React.useState(false);
+  const [data, setData] = React.useState(null);
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    if (!loading) {
+      if (data && successTrigger) {
+        successTrigger();
+      }
+      if (error && errorTrigger) {
+        errorTrigger();
+      }
+    }
+  }, [loading, data, error, successTrigger, errorTrigger]);
```

Comment:
I think the `if (!loading)` wrapping can be removed. The only time that `data` and `error` might potentially change is when `loading` changes from `true` to `false`. But even so, maybe keeping it in for clarity is best.

One problem with this is that if someone else uses this hook and passes in non-memoized functions then the effect will trigger every re-render. Could the triggers just be put in the `wcaRequest` below?